gamemode/sv_player.lua

```diff
@@ -203,81 +203,243 @@ function GM:PlayerDeathSound()
-		spawnPoints = table.Add( spawnPoints, ents.FindByClass( "aoc_spawnpoint" ) ) -- AOC Maps
-		spawnPoints = table.Add( spawnPoints, ents.FindByClass( "dys_spawn_point" ) ) -- Dystopia Maps
-		spawnPoints = table.Add( spawnPoints, ents.FindByClass( "info_player_coop" ) ) -- SYN Maps
-		spawnPoints = table.Add( spawnPoints, ents.FindByClass( "info_player_deathmatch" ) )
-	end
-
-
-	// recount
-	local Count = table.Count( spawnPoints )
-	
-	if ( Count == 0 ) then
-		Msg("[PlayerSelectSpawn] Error! No spawn points!\n")
-		return nil
+
+-- This is only a shallow copy.
+local function MergeTables(...)
+	local args = {...}
+	local new_table = {}
+	for _, tbl in ipairs(args) do
+		for _, value in ipairs(tbl) do
+			table.insert(new_table, value)
+		end
 	end
-	
-	local ChosenSpawnPoint = nil
-	
-	-- Try to work out the best, random spawnpoint
-	for i = 0, Count do
-	
-		ChosenSpawnPoint = table.Random( spawnPoints )
-
-		if ( ChosenSpawnPoint &&
-			ChosenSpawnPoint:IsValid() &&
-			ChosenSpawnPoint:IsInWorld() &&
-			ChosenSpawnPoint != pl:GetVar( "LastSpawnpoint" ) &&
-			ChosenSpawnPoint != self.LastSpawnPoint ) then
-			
-			if ( hook.Call( "IsSpawnpointSuitable", GAMEMODE, pl, ChosenSpawnPoint, i == Count ) ) then
-			
-				self.LastSpawnPoint = ChosenSpawnPoint
-				pl:SetVar( "LastSpawnpoint", ChosenSpawnPoint )
-				return ChosenSpawnPoint
-			
+
+	return new_table
+end
+
+
+-------------------------------
+-------------------------------
+-------------------------------
+
+-- Most of the following code from this comment block down to the next is from TTT.
+-- The code provides the functionality for automatically generating spawnpoints if
+-- necessary. Prophunters sort-of-not-really had a system for doing this but it
+-- basically just killed players every time or got them stuck in each other so it
+-- was not very useful. The TTT code has been tweaked to work better with Prophunters.
+
+
+-- Nice Fisher-Yates implementation, from Wikipedia
+local rand = math.random
+local function ShuffleTable(t)
+	local n = #t
+
+	while n > 2 do
+		-- n is now the last pertinent index
+		local k = rand(n) -- 1 <= k <= n
+		-- Quick swap
+		t[n], t[k] = t[k], t[n]
+		n = n - 1
+	end
+
+	return t
+end
+
+
+function GM:IsSpawnpointSuitable(ply, spwn, force, rigged)
+	if !IsValid(ply) || ply:Team() == 1 then return true end
+	if !rigged && (!IsValid(spwn) || !spwn:IsInWorld()) then return false end
+
+	-- spwn is normally an ent, but we sometimes use a vector for jury rigged
+	-- positions
+	local pos = rigged && spwn || spwn:GetPos()
+
+	if not util.IsInWorld(pos) then return false end
+
+	local blocking = ents.FindInBox(pos + Vector( -32, -32, 0 ), pos + Vector( 32, 32, 64 )) -- Changed from (-16, -16, 0) (16, 16, 64)
+
+	for _, blocking_ent in ipairs(blocking) do
+		if IsValid(blocking_ent) && blocking_ent:IsPlayer() && blocking_ent:Team() != 1 && blocking_ent:Alive() then
+			if force then
+				blocking_ent:Kill()
```

Comment:
Is this the code that does the too-many-players killing? Candidate for the message?

Reply:
Yep, this is it. I'm going to look into how I can send a message to admins only.